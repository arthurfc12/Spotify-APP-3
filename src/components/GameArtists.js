import axios from 'axios';
import React, { Component } from 'react';
import './App.css';
import { AiOutlineClose } from 'react-icons/ai';

class GameArtists extends Component {
	constructor(props) {
		super(props);
		this.state = {
			artista1: '',
			popularidade1: 0,
			artista2: '',
			popularidade2: 0,
			acertou: false,
			errou: false,
			refresh: true
		};

		this.handleFirstArtist = this.handleFirstArtist.bind(this);
		this.handleSecondArtist = this.handleSecondArtist.bind(this);
		this.handleRefresh = this.handleRefresh.bind(this);
	}

	componentWillMount() {
		this.setState({
			acertou: false,
			errou: false
		});
		const requestOptions = {
			headers: {
				Authorization: 'Bearer ' + this.props.token
			}
		};

		axios.get('https://api.spotify.com/v1/me/top/artists', requestOptions).then((response) => {
			var random1 = Math.floor(Math.random() * response.data.items.length);
			var random2 = Math.floor(Math.random() * response.data.items.length);

			while (response.data.items[random1].popularity === response.data.items[random2].popularity) {
				var random1 = Math.floor(Math.random() * response.data.items.length);
				var random2 = Math.floor(Math.random() * response.data.items.length);
			}
			this.setState({
				artista1: response.data.items[random1].name,
				popularidade1: response.data.items[random1].popularity,
				artista2: response.data.items[random2].name,
				popularidade2: response.data.items[random2].popularity
			});
			console.log(this.state.artista1);
			console.log(this.state.popularidade1);
			console.log(this.state.artista2);
			console.log(this.state.popularidade2);

			console.log(response.data);
		});
	}

	handleFirstArtist() {
		this.refs.btn1.setAttribute('disabled', 'disabled');
		this.refs.btn2.setAttribute('disabled', 'disabled');
		if (this.state.popularidade1 > this.state.popularidade2) {
			this.setState({
				acertou: true,
				errou: false
			});
		} else {
			this.setState({
				errou: true,
				acertou: false
			});
		}
	}

	handleSecondArtist() {
		this.refs.btn1.setAttribute('disabled', 'disabled');
		this.refs.btn2.setAttribute('disabled', 'disabled');
		if (this.state.popularidade2 > this.state.popularidade1) {
			this.setState({
				acertou: true
			});
		} else {
			this.setState({
				errou: true
			});
		}
	}
	handleRefresh() {
		window.location.reload(false);
		// this.forceUpdate();
		// console.log('entrou');
	}

	render() {
		return (
			<div className="container">
				<div>
					<h3 className="title"> Qual artista é mais famoso? </h3>
				</div>
				<div className="inRow">
					<button className="btn" ref="btn1" onClick={this.handleFirstArtist}>
						{' '}
						{this.state.artista1}
					</button>
					<div className="txt">
						{' '}
						<AiOutlineClose size={50} />{' '}
					</div>
					<button className="btn" ref="btn2" onClick={this.handleSecondArtist}>
						{' '}
						{this.state.artista2}
					</button>
				</div>
				<div className="Results">
					{this.state.acertou && (
						<div className="column">
							{' '}
							<div>Parabéns, você acertou!! </div>
							<div className="desc">
								{' '}
								{this.state.artista1} possui uma popularidade de {this.state.popularidade1}%{' '}
							</div>
							<div className="desc">
								{' '}
								{this.state.artista2} possui uma popularidade de {this.state.popularidade2}% {' '}
							</div>
							<button className="playAgain" onClick={this.handleRefresh}>
								{' '}
								<div> Ver outros Jogos</div>
							</button>
						</div>
					)}
					{this.state.errou && (
						<div className="column">
							{' '}
							<div> Que pena, você errou :( </div>
							<div className="desc">
								{' '}
								{this.state.artista1} possui uma popularidade de {this.state.popularidade1}%{' '}
							</div>
							<div className="desc">
								{this.state.artista2} possui uma popularidade de {this.state.popularidade2}%{' '}
							</div>
							<button className="playAgain" onClick={this.handleRefresh}>
								{' '}
								<div> Ver outros Jogos</div>
							</button>
						</div>
					)}
				</div>
			</div>
		);
	}
}

export default GameArtists;
