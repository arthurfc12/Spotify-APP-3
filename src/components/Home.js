import React from 'react';
import Game from './Game';
import Recommendation from './Recommendation';
import './App.css';
import Log from './Log.js';
import Button from '@material-ui/core/Button';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import Helmet from 'react-helmet';
import GameArtists from './GameArtists';
import ArtistRelated from './ArtistRelated';
import GameArtist from './GameArtist';

class Home extends React.Component {
  constructor(props) {
    super(props);
    const parametros = this.getHashParams();
    this.state = {
      token: '',
      refresh_token: '',
      page: 'game',
      drop: false,
      refresh: '',
      buttons: true,
    };
    if (this.state.token === '') {
      this.state.token = parametros.access_token;
      this.state.refresh_token = parametros.refresh_token;
    }
    this.handleClose = this.handleClose.bind(this);
    this.handleOpen = this.handleOpen.bind(this);
    this.handleChange1 = this.handleChange1.bind(this);
    this.handleChange2 = this.handleChange2.bind(this);
    this.handleChange3 = this.handleChange3.bind(this);
    this.refresh = this.refresh.bind(this);
    console.log(this.state);
    // const freshState = this.state;
  }

  componentDidMount() {
    const requestOptions = {
      method: 'GET',
      headers: {
        Authorization: 'Bearer ' + this.state.token,
      },
    };
    console.log(this.props);
    fetch('https://api.spotify.com/v1/me', requestOptions)
      .then((response) => response.json())
      .then((data) => this.setState({ userId: data }));
  }

  getHashParams() {
    var hashParams = {};
    var e,
      r = /([^&;=]+)=?([^&;]*)/g,
      q = window.location.hash.substring(1);
    e = r.exec(q);
    while (e) {
      hashParams[e[1]] = decodeURIComponent(e[2]);
      e = r.exec(q);
    }
    return hashParams;
  }

  handleChange1() {
    if (this.state.page === 'game') {
      this.setState({
        page: 'refresh',
        refresh: 'game',
        buttons: true,
        guessTheDate: false,
        whichArtist: false,
        guessTheArtist: false,
        artistRelated: false,
      });
    } else {
      this.setState({ page: 'game', buttons: true });
    }
  }

  handleChange2() {
    if (this.state.page === 'recommendation') {
      this.setState({
        page: 'refresh',
        refresh: 'recommendation',
      });
    } else {
      this.setState({ page: 'recommendation' });
    }
  }

  handleChange3() {
    if (this.state.page === 'log') {
      this.setState({
        page: 'refresh',
        refresh: 'log',
      });
    } else {
      this.setState({ page: 'log' });
    }
  }

  handleClose() {
    this.setState({ drop: false });
  }

  handleOpen() {
    this.setState({ drop: true });
  }

  refresh() {
    this.setState({ page: this.state.refresh });
  }

  render() {
    if (this.state.page === 'game') {
      return (
        <div className='gridContainer'>
          <Helmet>
            <title>Spotify App</title>
          </Helmet>
          <div className='barra1' />
          <div className='barra2'>
            <div className='bGroup'>
              <ButtonGroup
                variant='contained'
                color='primary'
                aria-label='contained primary button group'
              >
                <Button onClick={this.handleChange1}>Game</Button>
                <Button onClick={this.handleChange2}>Recomendação</Button>
                <Button onClick={this.handleChange3}>Histórico</Button>
              </ButtonGroup>
              <div>
                {this.state.buttons && (
                  <div>
                    <h1 style={{ color: '#FFFF', marginTop: 50 }}>
                      {' '}
                      Escolha seu Game!{' '}
                    </h1>
                    <div className='columnHome'>
                      <button
                        className='gameOpt'
                        onClick={() => {
                          this.setState({ whichArtist: true, buttons: false });
                        }}
                      >
                        {' '}
                        Qual artista é mais famoso?{' '}
                      </button>
                      <button
                        className='gameOpt'
                        onClick={() => {
                          this.setState({ guessTheDate: true, buttons: false });
                        }}
                      >
                        {' '}
                        Acerte a data de lançamento da música!{' '}
                      </button>
                      <button
                        className='gameOpt'
                        onClick={() => {
                          this.setState({
                            guessTheArtist: true,
                            buttons: false,
                          });
                        }}
                      >
                        {' '}
                        Acerte o Artista da música!{' '}
                      </button>
                      <button
                        className='gameOpt'
                        onClick={() => {
                          this.setState({
                            artistRelated: true,
                            buttons: false,
                          });
                        }}
                      >
                        {' '}
                        Busque artistas relacionados{' '}
                      </button>
                    </div>
                  </div>
                )}
                {this.state.whichArtist && (
                  <GameArtists token={this.state.token} />
                )}
                {this.state.guessTheDate && <Game token={this.state.token} />}
                {this.state.guessTheArtist && (
                  <GameArtist token={this.state.token} />
                )}
                {this.state.artistRelated && (
                  <ArtistRelated
                    id={this.state.userId}
                    token={this.state.token}
                  />
                )}
              </div>
            </div>
          </div>
          <div className='barra3' />
        </div>
      );
    } else if (this.state.page === 'recommendation') {
      return (
        <div className='gridContainer'>
          <Helmet>
            <title>Spotify App</title>
          </Helmet>
          <div className='barra1' />
          <div className='barra2'>
            <div className='bGroup'>
              <ButtonGroup
                variant='contained'
                color='primary'
                aria-label='contained primary button group'
              >
                <Button onClick={this.handleChange1}>Game</Button>
                <Button onClick={this.handleChange2}>Recomendação</Button>
                <Button onClick={this.handleChange3}>Histórico</Button>
              </ButtonGroup>
              <Recommendation token={this.state.token} />
            </div>
          </div>
          <div className='barra3' />
        </div>
      );
    } else if (this.state.page === 'log') {
      console.log(this.state.userId.id);
      return (
        <div className='gridContainer'>
          <Helmet>
            <title>Spotify App</title>
          </Helmet>
          <div className='barra1' />
          <div className='barra2'>
            <div className='bGroup'>
              <ButtonGroup
                variant='contained'
                color='primary'
                aria-label='contained primary button group'
              >
                <Button onClick={this.handleChange1}>Game</Button>
                <Button onClick={this.handleChange2}>Recomendação</Button>
                <Button onClick={this.handleChange3}>Histórico</Button>
              </ButtonGroup>
              <Log id={this.state.userId.id} token={this.state.token} />
            </div>
          </div>
          <div className='barra3' />
        </div>
      );
    } else {
      console.log(this.state.page);
      this.refresh();
      return <div className='screen'>Carregando</div>;
    }
  }
}

export default Home;
