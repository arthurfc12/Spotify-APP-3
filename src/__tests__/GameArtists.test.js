import React from 'react';
import GameArtists from '../components/GameArtists';
import Home from '../components/Home';

import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

it("should render 'which artist is more famous?' game", () => {
	const app = mount(<GameArtists />);
	expect(toJson(app)).toMatchSnapshot();
});
