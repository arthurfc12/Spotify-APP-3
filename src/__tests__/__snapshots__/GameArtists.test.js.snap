// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 'which artist is more famous?' game 1`] = `
<GameArtists>
  <div
    className="container"
  >
    <div>
      <h3
        className="title"
      >
         Qual artista Ã© mais famoso? 
      </h3>
    </div>
    <div
      className="inRow"
    >
      <button
        className="btn"
        onClick={[Function]}
      >
         
      </button>
      <div
        className="txt"
      >
         
        <AiOutlineClose
          size={50}
        >
          <IconBase
            attr={
              Object {
                "viewBox": "0 0 1024 1024",
              }
            }
            size={50}
          >
            <svg
              fill="currentColor"
              height={50}
              size={50}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width={50}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                key="0"
              />
            </svg>
          </IconBase>
        </AiOutlineClose>
         
      </div>
      <button
        className="btn"
        onClick={[Function]}
      >
         
      </button>
    </div>
    <div
      className="Results"
    />
  </div>
</GameArtists>
`;
